src/postalcode.js

```diff
@@ -0,0 +1,72 @@
+define([
+	"cldr",
+	"./core",
+	"./common/validate/cldr",
+	"./common/validate/default-locale",
+	"./common/validate/parameter-presence",
+	"./common/validate/parameter-type/string",
+
+	"cldr/event",
+	"cldr/supplemental"
+], function( Cldr, Globalize, validateCldr, validateDefaultLocale,
+	validateParameterPresence, validateParameterTypeString ) {
+
+/**
+ * .validatePostalCode( value )
+ *
+ * @value [String]
+ *
+ * Returns true if the value matches the postalcode pattern for the given locale,
+ * otherwise returns false
+ *
+ * Return the boolean.
+ */
+Globalize.validatePostalCode =
+Globalize.prototype.validatePostalCode = function( value, flag ) {
+	validateParameterPresence( value, "value" );
+	validateParameterTypeString( value, "value" );
+	return this.postalCodeValidator()( value, flag );
+};
+
+Globalize.postalCodeValidator =
+/**
+ * .postalCodeValidator()
+ *
+ * Return a validatePostalCode function (of the form below).
+ *
+ * fn( value )
+ *
+ * @value [String]
+ *
+ * Returns true if the value matches the postalcode pattern for the given locale,
+ * otherwise returns false
+ *
+ * Return the boolean.
+ *
+ */
+	Globalize.prototype.postalCodeValidator = function() {
+		var cldr, regex, pattern;
+
+	cldr = this.cldr;
+
+	validateDefaultLocale( cldr );
+
+	cldr.once( "get", validateCldr );
+	pattern = "^" + cldr.supplemental("postalCodeData/{region}") + "$";
```

Comment:
Yeap we surely do, well observed.

On Monday, March 16, 2015, hosup choi <notifications@github.com> wrote:

> In src/postalcode.js
> <https://github.com/jquery/globalize/pull/418#discussion_r26545177>:
>
> > + *
> > + * Returns true if the value matches the postalcode pattern for the given locale,
> > + * otherwise returns false
> > + *
> > + * Return the boolean.
> > + *
> > + */
> > +	Globalize.prototype.postalCodeValidator = function() {
> > +		var cldr, regex, pattern;
> > +
> > +	cldr = this.cldr;
> > +
> > +	validateDefaultLocale( cldr );
> > +
> > +	cldr.once( "get", validateCldr );
> > +	pattern = "^" + cldr.supplemental("postalCodeData/{region}") + "$";
>
> @rxaviers <https://github.com/rxaviers>
> Don't we need to check validation of "region" ? like
> "validateParameterTypeString(region, "region") "?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jquery/globalize/pull/418/files#r26545177>.
>


-- 
+55 (16) 98138-1582, +1 (415) 568-5854, skype: rxaviers
http://rafael.xavier.blog.br